eos-tech-support/eos-stage-flatpak

```diff
@@ -72,12 +72,17 @@ for remote in `flatpak remote-list --system | awk {'print $1'}` ; do
         # Update the collection ID if it's set
         collection_id=$(ostree --repo="$flatpak_repo" config get 'remote "'"$remote"'".collection-id' 2>/dev/null || echo "")
         if [ -n "$collection_id" ] ; then
+            # For info on the collection ID patterns, see
+            # https://phabricator.endlessm.com/w/software/ostree/infrastructure/#repository-stages
             if [ "$STAGE" == "prod" ] ; then
+                # change e.g. com.endlessm.Dev.Apps → com.endlessm.Apps or com.endlessm.Demo.Apps → com.endlessm.Apps
                 new_collection_id=`echo "$collection_id" | sed -e "s/Dev.//" | sed -e "s/Demo.//"`
             elif [ "$STAGE" == "dev" ] ; then
-                new_collection_id=`echo "$collection_id" | sed -e "s/Dev.//" | sed -e "s/Demo.//" | sed -e "s/endlessm./endlessm.Dev./"`
+                # change e.g. com.endlessm.Demo.Apps → com.endlessm.Dev.Apps or com.endlessm.Apps → com.endlessm.Dev.Apps
+                new_collection_id=`echo "$collection_id" | sed -e "s/Dev.//" | sed -e "s/Demo.//" | sed -e "s/com\.endlessm\./com.endlessm.Dev./"`
             elif [ "$STAGE" == "demo" ] ; then
-                new_collection_id=`echo "$collection_id" | sed -e "s/Dev.//" | sed -e "s/Demo.//" | sed -e "s/endlessm./endlessm.Demo./"`
+                # change e.g. com.endlessm.Dev.Apps → com.endlessm.Demo.Apps or com.endlessm.Apps → com.endlessm.Demo.Apps
+                new_collection_id=`echo "$collection_id" | sed -e "s/Dev.//" | sed -e "s/Demo.//" | sed -e "s/com\.endlessm\./com.endlessm.Demo./"`
```

Comment:
If escaping the `.`s` is necessary for `sed` (I can never remember and haven’t checked), then it needs to be done for all the other `.`s and `sed` invocations too.

Reply:
Yeah, here's proof that it's necessary:
```
$ echo "comMendlessm.Apps" | sed -e "s/com.endlessm/com.endlessm.Dev/"
com.endlessm.Dev.Apps
```